File: Creational/Builder/Cpp/CarDirector.cpp
// CarDirector.cpp
#include "CarDirector.h"

Car* CarDirector::buildLuxuryCar() {
  return (new CarBuilder())
          ->setEngine("V8 Turbo")
          ->setGPS(true)
          ->setSunroof(true)
          ->setSeats("Leather")
          ->build();
}

Car* CarDirector::buildEconomyCar() {
  return (new CarBuilder())
          ->setEngine("1.6L Petrol")
          ->setGPS(false)
          ->setSunroof(false)
          ->setSeats("Standard")
          ->build();
}

File: Creational/Builder/CSharp/CarDirector.cs
// CarDirector.cs

public class CarDirector {
  public Car BuildLuxuryCar() {
    return new Car.CarBuilder()
            .SetEngine("V8 Turbo")
            .SetGPS(true)
            .SetSunroof(true)
            .SetSeats("Leather")
            .Build();
  }

  public Car BuildEconomyCar() {
    return new Car.CarBuilder()
            .SetEngine("1.6L Petrol")
            .SetGPS(false)
            .SetSunroof(false)
            .SetSeats("Standard")
            .Build();
  }
}
